### PR TITLE
(PUP-10354) Add yardocs for Puppet::HTTP::Client

### PR DESCRIPTION
**Pull request by melissa**

### CODE REVIEW

File: lib/puppet/http/service.rb [hunk 1]
```diff
@@ -19,22 +37,48 @@ def self.create_service(client, session, name, server = nil, port = nil)
...
   end
 
+  #
+  # Open a connection using the given ssl context
+  #
+  # @param [Puppet::SSL::SSLContext] ssl_context (nil) optinal ssl context to connect with
+  #
   def connect(ssl_context: nil)
     @client.connect(@url, options: {ssl_context: ssl_context})
   end
```
Comment (melissa) [inline lib/puppet/http/service.rb]: Everything below this is `protected`, so I skipped documenting it

File: lib/puppet/http/client.rb [hunk 2]
```diff
@@ -144,6 +249,9 @@ def delete(url, headers: {}, params: {}, options: {})
     end
   end
 
+  #
+  # Close persistent connections in the pool
+  #
   def close
     @pool.close
   end
```
Comment (melissa) [inline lib/puppet/http/client.rb]: Everything below here is `private`, so I skipped documenting it. LMK if you want me to go back and add documentation

File: lib/puppet/http/service.rb (hunk 1 above)
Comment (joshcooper) [reply lib/puppet/http/service.rb]: 👍 to just public methods

File: lib/puppet/http/redirector.rb [hunk 3]
```diff
@@ -1,9 +1,28 @@
...
+  #
+  # Determine of the HTTP response code indicates a redirect
+  #
+  # @param [Net::HTTP] request request that received the response
+  # @param [Puppet::HTTP::Response] response
+  #
+  # @return [Boolean] true if the response code is 301, 302, or 307.
+  #
   def redirect?(request, response)
+    require 'pry'; binding.pry
```
Comment (joshcooper) [inline lib/puppet/http/redirector.rb]: There are a couple of `pry` leftovers

File: lib/puppet/http/response.rb [hunk 4]
```diff
@@ -1,37 +1,91 @@
+#
+# @api private
+#
+# Represents the response returned from the server from an HTTP request
+#
 class Puppet::HTTP::Response
-  attr_reader :nethttp, :url
```
Comment (melissa) [inline lib/puppet/http/response.rb]: I split these two up so they'd be rendered in the docs more appropriately

File: lib/puppet/http/redirector.rb [hunk 5]
```diff
@@ -14,6 +36,18 @@ def redirect?(request, response)
     end
   end
 
+  #
+  # @api private
+  #
+  # Implement the HTTP request redirection
+  #
+  # @param [Net::HTTP] request requet that has been redirected
```
Comment (joshcooper) [inline lib/puppet/http/redirector.rb]: `requet`

File: lib/puppet/http/resolver.rb [hunk 6]
```diff
@@ -1,12 +1,49 @@
+#
+# @api private
+#
+# Resolver base class. Each resolver represents a different stragegy for
+# resolving a service name into a list of candidate servers and ports.
```
Comment (joshcooper) [inline lib/puppet/http/resolver.rb]: `stragegy`

File: lib/puppet/http/resolver/server_list.rb [hunk 7]
```diff
@@ -1,4 +1,22 @@
+#
+# @api private
+#
+# Use the server_list setting to resolve a serivce. This resolver is only used
+# if server_list is set either on the command line or in the configuration file.
```
Comment (joshcooper) [inline lib/puppet/http/resolver/server_list.rb]: `serivce`

File: lib/puppet/http/response.rb [hunk 8]
```diff
@@ -1,37 +1,91 @@
...
+  #
+  # @return [String] Response body for the request
+  #
   def body
     @nethttp.body
   end
 
+  #
+  # @api private
+  #
```
Comment (joshcooper) [inline lib/puppet/http/response.rb]: Maybe mention this method streams the response body to the caller in chunks whereas the `body` method above returns the entire response body. Callers should use one or the other but not both for the same response.

File: lib/puppet/http/service/ca.rb [hunk 9]
```diff
@@ -1,12 +1,42 @@
...
 
+  #
+  # @api private
+  #
+  # @param [Puppet::HTTP::Client] client
+  # @param [Puppet::HTTP::Session] session
+  # @param [String] server (Puppet[:ca_server]) The default server used if the
+  #   settings resolver is used
+  # @param [Integer] port (Puppet[:ca_port])  The default port used if the
+  #   settings resolver is used
```
Comment (joshcooper) [inline lib/puppet/http/service/ca.rb]: Ah the word `default` is tripping me up. If the `server` and `port` arguments are given, then the caller wants to create a service for that explicit server/port. If the arguments are `nil` it means use the default values from settings for that kind of service.

File: lib/puppet/http/service/file_server.rb [hunk 10]
```diff
@@ -1,14 +1,55 @@
...
+  # @param [String] environment the name of the environment we are operating in
+  # @param [Symbol] links Can be one of either `:follow` or `:manage`, defines
+  #   how links are delt with.
+  # @param [String] checksum_type The digest algorithm used to verify the file.
+  #   Currently if fips is enabled, this defaults to `sha256`. Otherwise, it
+  #   defaults to `md5`.
+  # @param [Symbol] source_permissions Can be one of `:use`, `:use_when_creating`,
+  #   or `:ignore`. This parameter tells the server if it should include the
+  #   file permissions in the report. If set to `:ignore`, the server will return
+  #   default permissions.
```
Comment (joshcooper) [inline lib/puppet/http/service/file_server.rb]: The word `report` threw me off. Maybe "should include in the response"?

**Comment by joshcooper** [comment]
Looks great, thanks for adding these!

File: lib/puppet/http/redirector.rb (hunk 5 above)
Comment (melissa) [reply lib/puppet/http/redirector.rb]: updated

File: lib/puppet/http/resolver.rb (hunk 6 above)
Comment (melissa) [reply lib/puppet/http/resolver.rb]: updated

File: lib/puppet/http/resolver/server_list.rb (hunk 7 above)
Comment (melissa) [reply lib/puppet/http/resolver/server_list.rb]: updated

**Comment by melissa** [comment]
@clairecadman I'm not great at grammar or spelling, so it'd be awesome if you could review these documentation changes!

**Comment by joshcooper** [comment]
@clairecadman I'm going to merge this, but feel free to comment on the PR and we can submit changes afterwards.

**Pull request merged**